framework/include/utils/InputParameters.h

```diff
@@ -728,6 +739,9 @@ class InputParameters : public Parameters
   /// A list of parameters declared as controllable
   std::set<std::string> _controllable_params;
 
+  /// The reserved option names for a parameter
+  std::map<std::string, std::set<std::string>> _reserved_values;
```

Comment:
Yes, the copy constructors uses the assignment operator which has been updated. Same for the += operator.